### ISSUE TITLE
histogram , logarithmic scale , negative values.

### ISSUE THREAD

**Issue by Aryo-Zare** [matplotlib/matplotlib#26415]
### Problem

I would like to tackle an issue by plotting the histogram of a data with negative values in logarithmic scale.
Hence there are 3 factors : data with negative values , histogram , logarithmic scale.

Seaborn package can plot a histogram with logarithmically sized bins. Hence bins visually seem equal in axes with logarithmics scale. But seaborn can not plot negative values.
I contrast, matplotlib can plot negative values in logarithmic scale (‘symlog’), but can not display a proper histogram (the bin sizes are not correspondingly logarithmically sized, they are linearly sized).

I suggest initially defining the bin sizes in logscale & then assigning them as bins.
I would like to have an object-oriented parameter in the hist function to do this.
The frame of the method :

### Proposed solution

```python
data = df[‘column’]
nb = 60  # number of desired bins

logbins = np.logspace( np.log10( data.abs().min() ) , np.log10( data.abs().max() ) , int(nb/2) )
logbins_2 = np.concatenate((-1*np.flip(logbins) , logbins))
logbins_3 = np.sort(logbins_2) 

plt.hist( data , bins=logbins_3 )
plt.xscale('symlog' , linthresh=0.001)  
```

I tested this and it functions correctly.
I’m not much familiar with object-oriented programming & using git. Hence it would be good that some other volunteer here continue the job by creating a parameter for the hist function : for example   

```python
plt.hist( data , symlogbins=True )
```

**Comment by jklymak** [comment]
I don't think we'd accept such a change. We use np.hist and want to avoid doing too much different from what it does. As you note, we allow you to pass your bins in manually, so that works for users with specialized needs.

**Comment by timhoffm** [comment]
I agree with @jklymak Our focus is on visualization, not on data processing. If logarithmic scale on histograms are important enough, that should go into the algorithmic code, i.e. `numpy.histogram`. If you can convince `numpy` to do the logic, we'll pick that interface up.

**Comment by Aryo-Zare** [comment]
Alright. I had not thought about numpy as the main processor of data in matplotlib.hist function.
I’ll contact the numpy community & will return here with the outcome.
